OpenRA.Mods.Common/Activities/Air/ReturnToBase.cs

```diff
@@ -108,7 +108,10 @@ public override Activity Tick(Actor self)
 
 				self.CancelActivity();
 				if (nearestAfld != null)
-					return ActivityUtils.SequenceActivities(new Fly(self, Target.FromActor(nearestAfld)), new FlyCircle(self));
+					return ActivityUtils.SequenceActivities(
+						new Fly(self, Target.FromActor(nearestAfld), WDist.Zero, plane.Info.WaitDistanceFromResupplyBase),
+						new FlyCircleTimed(plane.Info.NumberOfTicksToVerifyAvailableAirport, self),
+						new ReturnToBase(self));
```

Comment:
Use `this` instead of creating a new instance. I had better results with that change. The activity queue probably gets lost otherwise.